automation_tools/utils.py

```diff
@@ -79,3 +82,36 @@ def update_packages(*args, **kwargs):
         quiet=kwargs.get('quiet', False),
         warn_only=kwargs.get('warn_only', False),
     )
+
+
+def get_discovery_image():
+    """ Task for getting unattended foreman-discovery ISO image
+    :return: foreman-discovery-image iso under /var/lib/libvirt/images/
+    """
+    url = os.environ.get('BASE_URL') + '/Packages/'
+    try:
+        from BeautifulSoup import BeautifulSoup
+    except Exception:
+        subprocess.call(['pip', 'install', 'BeautifulSoup'])
+        from BeautifulSoup import BeautifulSoup
+
+    soup = BeautifulSoup(urllib2.urlopen(url).read())
+    for link in soup.findAll('a'):
+        if 'foreman-discovery-image' in link.string:
+            discovery_image = link.string
+    try:
+        run("wget -O /tmp/" + discovery_image + " " + url + discovery_image)
+        run('rpm -Uvh /tmp/' + discovery_image)
```

Comment:
just thinking instead of installing rpm, lets try to extract the files through rpm2cpio, e.g. `cd tmpdir;  rpm2cpio discovery.rpm | cpio -idmv`